com.unity.multiplayer.mlapi/Runtime/SceneManagement/NetworkSceneManager.cs

```diff
@@ -389,6 +389,13 @@ private void OnSceneUnloadClient(Guid switchSceneGuid, Stream objectStream)
                     }
 
                     var networkObject = m_NetworkManager.SpawnManager.CreateLocalNetworkObject(true, prefabHash, ownerClientId, parentNetworkId, position, rotation);
+                    if (networkObject == null)
+                    {
+                        //This will prevent one misconfigured issue (or more) from breaking the entire loading process.
+                        Debug.LogError($"Failed to spawn NetowrkObject for Hash {prefabHash}, ignoring and continuing to load.");
```

Comment:
Do you think we should error and continue or assert?  Are we really in a recoverable state here or have we put the user in a situation where we haven't hit an exception but the state is hopelessly messed up?

Reply:
This we can discuss further in the Thursday meeting, but the big different between allowing the exception to happen vs. gracefully handling each specific case is:
1.) With an exception, you break out of the current stack scope and if there were any remaining and valid NetworkObjects to load you will never load them (i.e. bug reports can vary widely for the same issue focused around one thing failing) and it becomes a complete blocker for developers as they can't test anything else until that one specific issue is resolved.
2.) "Gracefully" handling each specific case will only expose the specific NetworkObjects causing the issue.  This helps to greatly narrow down what is broken and what is still working as opposed to just stopping the entire loading process when you encounter 1 NetworkObject that has an issue (i.e. there could be many other valid NetworkObjects entries in the stream and a bug report could be based around any one of the valid NetworkObjects that never loaded on the client side). This also only blocks on the NetworkObject(s) failing, but doesn't block a developer from being able to test everything that does work.

I picked the later of the two as I did find it annoying to not be able to test other things if 1 aspect of MLAPI was broken (specifically soft synchronization issues).  More than likely with our new GlobalObjectIdHash pattern this type of thing won't happen.  

I used a similar approach when first loading the NetworkPrefab list at runtime:
- Encounter and issue with a NetworkPrefab at runtime (i.e. null or the like)
- log a warning or error depending upon the severity
- remove it from the list of NetworkPrefabs
- continue loading process

_There is no harm in this and it allows a developer to at least test what is valid at that moment, and then they can circle back to fixing any errors/warnings next editor session._

Reply:
I think this is well said and thought out.  I think the sticking point for me is I don't have faith people look at the console for errors, but I know we'll get their attention if it doesn't run. I mean if we really wanted the game to defensively cope and keep on trucking and did some kind of scheme like spawn a built-in-stand-in-can't-fail prefab, then I could live with it.  But I think masking errors ultimately makes life worse for users.

Reply:
both exception logs and error logs appear in the Unity console logs with the exact same UI. if somebody doesn't want to pay attention to console error logs, they are also not paying attention to exception logs — UI-wise there is no difference. I agree with Noel's approach here where we just log an error to console but continue loading/serializing the rest. yeah, we can discuss internally later.

Reply:
This is resolved/removed and will be handled via [PR-756](https://github.com/Unity-Technologies/com.unity.multiplayer.mlapi/pull/756).